ui/pages/setup_passkey.tsx

```diff
@@ -80,27 +89,107 @@ const SetupPasskey: NextPage = () => {
           },
         })
         .then(() => {
-          window.location.href = "./setup_passkey";
+          setFlow(undefined);
+          setLoadingKeysFlow(flow);
         })
         .catch(handleFlowError("settings", setFlow));
     }
     return Promise.resolve();
   };
 
+  const renderFlow = flow ?? loadingKeysFlow;
+  const existingKeys: UiNode[] = [];
+
   const webauthnFlow = {
-    ...flow,
+    ...renderFlow,
     ui: {
-      ...flow?.ui,
-      nodes: flow?.ui.nodes.filter(({ group }) => {
-        return group === "webauthn" || group === "default";
-      }),
+      ...renderFlow?.ui,
+      nodes: renderFlow?.ui.nodes
+        .filter((node) => {
+          if (node.meta.label?.id === ORY_LABEL_ID_REMOVE_SECURITY_ID) {
+            existingKeys.push(node);
+            return false;
+          }
+          return node.group === "webauthn" || node.group === "default";
+        })
+        .map((node) => {
+          if (node.meta.label?.id === ORY_LABEL_ID_NAME_OF_THE_SECURITY_KEY) {
+            node.meta.label.text = "Security key name";
+          }
+          return node;
+        }),
     },
   } as SettingsFlow;
 
   return (
-    <PageLayout title="Set up a passkey login method">
-      {flow ? (
-        <Flow onSubmit={handleSubmit} flow={webauthnFlow} />
+    <PageLayout title="Add a security key">
+      {webauthnFlow ? (
+        <>
+          <Flow onSubmit={handleSubmit} flow={webauthnFlow} />
```

Comment:
currently if I add a key with the same name, it is possible to do so. Is this something we should prevent the user from doing?

![Screenshot from 2024-11-18 16-37-57](https://github.com/user-attachments/assets/aabf103b-bf02-401e-a923-e176e64e9e8b)

Reply:
Any validation like deduplication needs be done in the backend as frontend validation can always be messed with by a knowledgeable user. @BarcoMasile @nsklikas @natalian98 can you add deduplication of the passkey names?

Reply:
Agreed that validation should be performed on the backend, but I do think in the UI we can improve user experience by blocking the user from submitting a duplicate key name. That being said, from what I can see, this might require some non-trivial rearrangements in the `NodeInputButton` as it is a generalised component for adding keys and performing other actions. Perhaps this can be a follow up task in the UI?

Reply:
@edlerd we'd have to do some investigation on this, I'm not sure about the answer. Let's add an issue for this.
Validation should be done in both places if possible, so we don't let users do "wrong things" when we can avoid it